src/gui/transferlistmodel.cpp

```diff
@@ -505,6 +517,10 @@ QVariant TransferListModel::internalValue(const BitTorrent::Torrent *torrent, co
         return torrent->distributedCopies();
     case TR_TOTAL_SIZE:
         return torrent->totalSize();
+    case TR_INFOHASH_V1:
+        return torrent->infoHash().v1().toString();
+    case TR_INFOHASH_V2:
+        return torrent->infoHash().v2().toString();
```

Comment:
Ok, I just have a question because Qt is new to me.

This function should return a `QVariant` so, if I understand the doc, can contain any Q* types and few std types.
`v1()` and `v2()` return `SHA1Hash` and `SHA256Hash` which are in turn `Digest32`. So, if I want QVariant to hold these objects, I need to add `Q_DECLARE_METATYPE(SHA256Hash)` and  `Q_DECLARE_METATYPE(SHA1Hash)` to this class header file? Is there side effects of doing this?

Reply:
>I need to add Q_DECLARE_METATYPE(SHA256Hash) and Q_DECLARE_METATYPE(SHA1Hash) to infohash.h header file? 

Yes. 

>Is there side effects of doing this?

You are not the first one using `Q_DECLARE_METATYPE()`. We use it already. Just add the declaration under the type aliases in infohash.h.

Reply:
It was just to know if adding this could break existing things or implied that I add to change some other stuffs in the code ;)